src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3023,28 +3114,50 @@ export class StudyViewPageStore {
 
     public getCustomChartDataCount(chartMeta: ChartMeta) {
         let uniqueKey: string = chartMeta.uniqueKey;
+        
         if (!this.customChartsPromises.hasOwnProperty(uniqueKey)) {
             this.customChartsPromises[uniqueKey] = remoteData<ClinicalDataCountWithColor[]>({
-                await: () => [this.selectedSamples],
+                await: () => {
+                    return _.includes([UniqueKey.WITH_MUTATION_DATA, UniqueKey.WITH_CNA_DATA],uniqueKey) ? [this.molecularProfileSampleCounts, this.selectedSamples] : [this.selectedSamples];
+                },
                 invoke: async () => {
-                    const result = _.reduce(this.selectedSamples.result, (acc, sample) => {
-                        const findCase = _.find(this._customChartsSelectedCases.get(uniqueKey), (selectedCase:CustomChartIdentifierWithValue) => selectedCase.sampleId === sample.sampleId);
-                        let value =  'NA';
-                        if(findCase !== undefined) {
-                            value =  findCase.value;
-                        }
+                    let result: { [id: string]: ClinicalDataCount } = {};
```

Comment:
we should extract the code in this invoke to pure helper function and have a test for it.